src/contracts/interfaces/IReleaseManager.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@eigenlayer-contracts/src/contracts/interfaces/IPermissionController.sol";
+
+interface IReleaseManagerErrors {
+    /// @notice Thrown when an AVS is not registered
+    error AVSNotRegistered();
+    /// @notice Thrown when an AVS is already registered
+    error AVSAlreadyRegistered();
+    /// @notice Thrown when a release is not found
+    error ReleaseNotFound();
+    /// @notice Thrown when trying to publish with an invalid deadline
+    error InvalidDeadline();
+    /// @notice Thrown when caller lacks required permissions
+    error Unauthorized();
+    /// @notice Thrown when attempting to deprecate an already deprecated release
+    error AlreadyDeprecated();
+    /// @notice Thrown when digest is invalid
+    error InvalidDigest();
+}
+
+interface IReleaseManagerEvents {
+    /// @notice Emitted when a release is published
+    event ReleasePublished(
+        address indexed avs,
+        string indexed version,
+        bytes32 indexed digest,
+        string registryUrl,
+        uint256 deploymentDeadline
+    );
+
+    /// @notice Emitted when a release is deprecated
+    event ReleaseDeprecated(
+        address indexed avs,
+        bytes32 indexed digest
+    );
+
+    /// @notice Emitted when an AVS registers
+    event AVSRegistered(address indexed avs);
+
+    /// @notice Emitted when an AVS deregisters
+    event AVSDeregistered(address indexed avs);
+}
+
+interface IReleaseManager is IReleaseManagerErrors, IReleaseManagerEvents {
+    struct PublishedRelease {
+        bytes32 digest;
+        string registryUrl;
+        string version;
+        uint256 deploymentDeadline;
+        uint256 publishedAt;
+    }
```

Comment:
Small nit: Move this to a `IReleaseMangerTypes` interface above and inherit that here. Take a look at how the Mailbox does it.